pkg/controllers/clusterresourceplacementeviction/controller_intergration_test.go

```diff
@@ -38,6 +41,99 @@ const (
 	consistentlyInterval = time.Millisecond * 500
 )
 
+var _ = Describe("Test ClusterResourcePlacementEviction complete metric", Ordered, func() {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	evictionName := fmt.Sprintf(evictionNameTemplate, GinkgoParallelProcess())
+	var customRegistry *prometheus.Registry
+
+	BeforeEach(func() {
+		// Create a test registry
+		customRegistry = prometheus.NewRegistry()
+		Expect(customRegistry.Register(metrics.FleetEvictionStatus)).Should(Succeed())
+		// Reset metrics before each test
+		metrics.FleetEvictionStatus.Reset()
+		// emit incomplete eviction metric to simulate eviction failed once.
+		metrics.FleetEvictionStatus.WithLabelValues(evictionName, "false", "unknown").SetToCurrentTime()
+	})
+
+	AfterEach(func() {
+		ensureAllBindingsAreRemoved(crpName)
+		ensureEvictionRemoved(evictionName)
+		ensureCRPRemoved(crpName)
```

Comment:
no need to remove registry or GC customRegistry?

Reply:
The registry is a new object in each BeforeEach and it's scope is confined to the Describe block, ungerstering eviction metric as part of cleanup